### ISSUE TITLE
Detecting and Using Direct Pairs

### ISSUE THREAD

**Issue by ghost** [ccxt/binance-trade-bot#61]
Is there a way for me to use the script to only trade between BTC <-USDT -> ETH?

Would this entail that I edit the supported coins file and just input those coins in there and remove the rest?

**Comment by remriel** [comment]
Yes 

`supported_coin_list = [u'BTC', u'ETH']`

**Comment by abanchev** [comment]
Do not do it like this, there is a direct ETHBTC pair, so you don't need to pass through USD (and lose money in fees and potential slippage). You will need to write your own function to do the trade directly if one of the coins is BTC

**Comment by remriel** [comment]
I think the reason to use USDT is that it's more stable and you can accumulate more BTC/ETH this way.

**Comment by edeng23** [comment]
@abanchev  I agree, if a direct pair exists the bot should make a direct hop. Since I never ran the bot with ETH and BTC I didn't bother to add this logic. I'll do it as soon as I'm done with all my tests in Uni, or let someone else make a PR.

**Comment by BenQoder** [comment]
I will attempt making a PR to this effect and see how far I can go.

**Comment by BenQoder** [comment]
I will attempt making a PR to this effect and see how far I can go.

**Comment by BenQoder** [comment]
I will attempt making a PR to this effect and see how far I can go.

**Comment by CTetford** [comment]
Anyone willing to test, feel free:
https://github.com/edeng23/binance-trade-bot/pull/84

**Comment by ghost** [comment]
I'm going it a go now and will report my findings.

Is there a way to change the script so that it will trade the coins when there is certain percentage difference between the coins when up against the usdt?

**Comment by CTetford** [comment]
Yeah it's definitely possible. I've added in a "minimum profit" percentage variable to my script. I ran some backtesting and found 3-4% profit gives the best results, but it was with what limited data I could find.

I set up a simulation version where it uses price info from here:
http://www.cryptodatadownload.com/data/gemini/

There's per-minute pricing for BTC, LTC, ETH, and ZEC.

**Comment by ghost** [comment]
Nice. I got the script and it is running now. I'll provide feedback in the morning

**Comment by Cactii1** [comment]
This is really important because when using USDT there is double the tx fees because it's two trades that are executed to get from one coin to the other.

**Comment by remriel** [comment]
I tried doing this but I'm getting errors. I added BTC and ETH to the supported coin list and set the bridge to ETHBTC.

2021-04-05 17:26:24,355 - crypto_trader_logger - INFO - Started
2021-04-05 17:26:25,261 - crypto_trader_logger - INFO - Creating database schema if it doesn't already exist
2021-04-05 17:26:25,679 - crypto_trader_logger - INFO - Initializing <BTC> vs <ETH>
2021-04-05 17:26:25,680 - crypto_trader_logger - INFO - Skipping initializing BTCETHBTC, symbol not found
2021-04-05 17:26:25,680 - crypto_trader_logger - INFO - Initializing <ETH> vs <BTC>
2021-04-05 17:26:25,681 - crypto_trader_logger - INFO - Skipping initializing ETHETHBTC, symbol not found
2021-04-05 17:26:31,137 - crypto_trader_logger - INFO - Skipping scouting... current coin BTCETHBTC not found
2021-04-05 17:26:36,577 - crypto_trader_logger - INFO - Skipping scouting... current coin BTCETHBTC not found
